smartsim/settings/common.py

```diff
@@ -0,0 +1,49 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2024, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from __future__ import annotations
+
+import typing as t
+
+from smartsim.log import get_logger
+
+IntegerArgument = t.Dict[str, t.Optional[int]]
+StringArgument = t.Dict[str, t.Optional[str]]
+
+logger = get_logger(__name__)
+
+
+def set_check_input(key: str, value: t.Optional[str]) -> None:
+    if not isinstance(key, str):
+        raise TypeError(f"Key '{key}' should be of type str")
+    if not isinstance(value, (str, type(None))):
+        raise TypeError(f"Value '{value}' should be of type str or None")
+    if key.startswith("-"):
+        key = key.lstrip("-")
+        logger.warning(
+            "One or more leading `-` characters were provided to the run argument. \
+            Leading dashes were stripped and the arguments were passed to the run_command."
```

Comment:
I'm guessing this message was supposed to read
```suggestion
            "One or more leading `-` characters were provided to the run argument.\n"
            "Leading dashes were stripped and the arguments were passed to the run_command."
```
to avoid the multiple spaces at the start of the next line. Can you update while we are here?

Reply:
fixed